klippa/src/lib.rs

```diff
@@ -600,6 +800,50 @@ pub trait Subset {
     ) -> Result<(), SubsetError>;
 }
 
+// A helper trait providing a 'subset' method for various subtables that have no associated tag
+pub(crate) trait SubsetTable {
+    /// Subset this table and write a subset version of this table into serializer
+    fn subset(&self, plan: &Plan, s: &mut Serializer) -> Result<(), SerializeErrorFlags>;
+}
+
+// A trait for subtables that require additional data in order to be subsetted
+pub(crate) trait SubsetArgs {
+    type ArgsForSubset: Default;
+}
+
+pub(crate) trait SubsetTableWithArgs: SubsetArgs {
```

Comment:
is the trait dependency necessary? Do you sometimes impl `SubsetArgs` but not `SubsetTableWithArgs`?